modules/uhttp/http.go

```diff
@@ -83,15 +83,26 @@ type Config struct {
 // GetHandlersFunc returns a slice of registrants from a service host
 type GetHandlersFunc func(service service.Host) []RouteHandler
 
-// New returns a new HTTP module
-func New(hookup GetHandlersFunc, options ...ModuleOption) service.ModuleCreateFunc {
-	return func(mi service.Host) (service.Module, error) {
-		return newModule(mi, hookup, options...)
-	}
+// New returns a new HTTP ModuleProvider.
+func New(hookup GetHandlersFunc, options ...ModuleOption) service.ModuleProvider {
+	return &moduleProvider{hookup, options}
+}
+
+type moduleProvider struct {
+	hookup  GetHandlersFunc
+	options []ModuleOption
```

Comment:
Oh I see what you mean. These are `yarpc.ModuleOptions`, not `service.ModuleOptions`, that's why they are here.